src/managers/input/InputManager.cpp

```diff
@@ -601,8 +601,12 @@ void CInputManager::processMouseDownNormal(wlr_pointer_button_event* e) {
     const auto mouseCoords = g_pInputManager->getMouseCoordsInternal();
     const auto w           = g_pCompositor->vectorToWindowIdeal(mouseCoords);
 
-    if (w && !m_bLastFocusOnLS && w->checkInputOnDecos(INPUT_TYPE_BUTTON, mouseCoords, e))
-        return;
+    if (w && !m_bLastFocusOnLS) {
```

Comment:
w is ideal, but does ideal include decos? I don't think so... so we would get the window behind, no?

Reply:
`CCompositor::vectorToWindowIdeal()` uses `CWindow::getWindowInputBox()` which does include, which does include all extents (not just reserved)
so we do get the proper window, which is the same that is checked for input on decos with the same logic

on another note, `CWindow::getWindowInputBox()` includes non reserved extents 
this makes it so clicking on shadows (for example) focuses the window
is this desired or should it include only the outer reserved extents?

Reply:
ouh, I thought ideal uses vecposition and vecsize. Doh.

Ideal should be vecpos and vecsize. That's it.

Reply:
should I make a clone of `CCompositor::vectorToWindowIdeal()` and use that here, so `CCompositor::vectorToWindowIdeal()` gets changed in the future
or should we try to unify the existing get window functions and adding extra parameters to get the same functionality?

Reply:
we could unify them. Ideal should only use vecpos vecsize (it's for determining windows' "input boxes")

Reply:
created 2 new functions with the purpose of achieving all possible queries
at the moment `CCompositor::vectorToWindowUnified` is just `CCompositor::vectorToWindowIdeal` with some extra logic to exclude/restrict to float and uses `CWindow::getWindowBoxUnified` instead of `CWindow::getWindowInputBox`